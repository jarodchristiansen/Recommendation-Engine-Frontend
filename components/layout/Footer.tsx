import Link from "next/link";

const Footer = () => {
  const currentYear = new Date().getFullYear();

  const links = [
    { id: 1, text: "home", link: "/" },
    { id: 2, text: "dashboard", link: "/dashboard" },
    { id: 3, text: "recommendations", link: "/recommendations" },
    // { id: 4, text: "about", link: "/about" },
    // { id: 5, text: "contact", link: "/contact" },
    { id: 6, text: "Sign In/Up", link: "/auth" },
  ];

  return (
    <footer className="bg-black text-white py-8">
      <div className="container mx-auto ">
        {/* Main Navigation Links */}
        <div className="flex flex-col items-center justify-center text-center gap-6 md:flex-row md:gap-8 border-b border-gray-700 pb-4 mb-4">
          {links.map(({ id, text, link }) => (
            <Link href={link} key={id}>
              <span className="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out">
                {text}
              </span>
            </Link>
          ))}
        </div>

        {/* Copyright Info */}
        <div className="text-center text-gray-400">
          <p>&copy; {currentYear} All rights reserved.</p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
