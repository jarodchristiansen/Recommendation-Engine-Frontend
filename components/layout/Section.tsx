import Button from "@/components/layout/Button";

type SectionProps = {
  title: string;
  toggle: boolean;
  setToggle: (toggle: boolean) => void;
  children: React.ReactNode;
};

/* Reusable Section Component */
const Section = ({ title, toggle, setToggle, children }: SectionProps) => (
  <section className="mb-12">
    <div className="flex items-center justify-between">
      <h2 className="text-2xl font-bold">{title}</h2>
      <Button onClick={() => setToggle(!toggle)}>
        {toggle ? "Hide" : "Show"}
      </Button>
    </div>
    {toggle && <div className="mt-6">{children}</div>}
  </section>
);

export default Section;
