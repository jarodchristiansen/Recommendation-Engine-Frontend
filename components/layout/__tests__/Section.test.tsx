import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Footer from "@/components/layout/Footer"; // Adjust path as necessary
import Section from "../Section";

describe("Footer Component", () => {
  //   beforeEach(() => {
  //     render(<Footer />);
  //   });

  // snapshhot test

  const props = {
    title: "Test Section",
    toggle: true,
    setToggle: jest.fn(),
    children: <p>Test Children</p>,
  };

  it("matches the snapshot of the Footer", () => {
    const { container } = render(<Section {...props} />);

    // Create a snapshot of the rendered Footer
    expect(container).toMatchSnapshot();
  });
});
