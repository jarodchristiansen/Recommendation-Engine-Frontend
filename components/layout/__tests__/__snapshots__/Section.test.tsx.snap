// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot of the Footer 1`] = `
<div>
  <section
    class="mb-12"
  >
    <div
      class="flex items-center justify-between"
    >
      <h2
        class="text-2xl font-bold"
      >
        Test Section
      </h2>
      <button
        class="transition-transform transform hover:scale-105 shadow-lg rounded-full font-semibold focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 bg-red-500 hover:bg-red-600 text-white px-6 py-3 text-md"
        type="button"
      >
        Hide
      </button>
    </div>
    <div
      class="mt-6"
    >
      <p>
        Test Children
      </p>
    </div>
  </section>
</div>
`;
